i18n-helpers/src/bin/markdown-gettext.rs

```diff
@@ -0,0 +1,248 @@
+use anyhow::{anyhow, bail, Context};
+use mdbook_i18n_helpers::translate;
+use polib::catalog::Catalog;
+use polib::po_file;
+use std::fs::File;
+use std::io::{BufReader, Read, Write};
+use std::path::PathBuf;
+use std::{env, fs};
+
+fn split_argument(arg: &String) -> Option<String> {
+    let splitted: Vec<_> = arg.split('=').collect();
+    if splitted.len() == 2 {
+        return Some(splitted[1].to_string());
+    }
+
+    None
+}
+
+fn build_catalog(lang_option: Option<String>) -> anyhow::Result<Catalog> {
+    match lang_option {
+        Some(lang_path) => {
+            let pot_path = PathBuf::from(&lang_path);
+
+            if pot_path.exists() {
```

Comment:
It is normally easier to just call the I/O function and let that one raise an error if something doesn't exist, cannot be read, is the wrong type, etc... 

Notice that the file could disappear under your feet in the split second between you calling `pot_path.exists()` and calling `po_file::parse(&pot_path)` below :slightly_smiling_face:  This implies that you need to somehow deal with the file not existing from the `parse` call — which you do nicely by returning an error.

Removeing the `if` statement will also save you a level of nesting and simplify the code a bit.